server/src/test/java/org/elasticsearch/common/util/concurrent/SizeBlockingQueueTests.java

```diff
@@ -47,18 +49,25 @@ public void testQueueSize() throws InterruptedException {
         }
 
         final CountDownLatch latch = new CountDownLatch(1);
+        final CyclicBarrier barrier = new CyclicBarrier(2);
 
-        final AtomicBoolean spin = new AtomicBoolean(true);
         final AtomicInteger maxSize = new AtomicInteger();
 
         // this thread will repeatedly poll the size of the queue keeping track of the maximum size that it sees
+        final int iterations = 1 << 16;
         final Thread queueSizeThread = new Thread(() -> {
             try {
                 latch.await();
```

Comment:
Looking at this I was wondering if the latch was needed with the introduction of the barrier, it can probably be removed.